benchmark/solver/solver.cpp

```diff
@@ -216,10 +219,53 @@ const std::map<std::string, std::function<std::unique_ptr<gko::LinOpFactory>(
          }}};
 
 
+void write_precond_info(const gko::LinOp *precond,
+                        rapidjson::Value &precond_info,
+                        rapidjson::MemoryPoolAllocator<> &allocator)
+{
+    if (auto jacobi =
+            dynamic_cast<const gko::preconditioner::Jacobi<etype> *>(precond)) {
+        // extract block sizes
+        auto bdata = jacobi->get_parameters().block_pointers.get_const_data();
+        add_or_set_member(precond_info, "block_sizes",
+                          rapidjson::Value(rapidjson::kArrayType), allocator);
+        for (auto i = 0u; i < jacobi->get_num_blocks(); ++i) {
```

Comment:
`Jacobi::get_num_blocks()` returns gko::size_type, which is 64 bit, whereas `i` will of type `unsigned int` (because it is the type of `0u`).

If you want to stick with auto:
```suggestion
        for (auto i = gko::size_type{0}; i < jacobi->get_num_blocks(); ++i) {
```
Or actually adaptive but harder to read (I hope it works like that, did not test it):
```suggestion
        for (auto i = decltype(jacobi->get_num_blocks()){0}; i < jacobi->get_num_blocks(); ++i) {
```